enterprise/internal/database/perms_store.go

```diff
@@ -1588,29 +1590,44 @@ AND repo.id NOT IN
 	return ids, nil
 }
 
-func (s *permsStore) UserIDsWithOldestPerms(ctx context.Context, limit int) (map[int32]time.Time, error) {
+// UserIDsWithOldestPerms lists the users with the oldest synced perms, limited
+// to limit. If age is non-zero, users that have synced within "age" since now
+// will be filtered out.
+func (s *permsStore) UserIDsWithOldestPerms(ctx context.Context, limit int, age time.Duration) (map[int32]time.Time, error) {
+	cutoffClause := sqlf.Sprintf("TRUE")
+	if age > 0 {
+		cutoff := s.clock().Add(-1 * age)
+		cutoffClause = sqlf.Sprintf("(perms.synced_at IS NULL OR (perms.synced_at IS NOT NULL AND perms.synced_at < %s))", cutoff)
```

Comment:
```suggestion
		cutoffClause = sqlf.Sprintf("(perms.synced_at IS NULL OR perms.synced_at < %s)", cutoff)
```

I am not sure if this condition is absolutely necessary, but I think having it doesn't hurt. The suggestion above is not a real suggestion but just to show the diff of what I'm referring to.

Reply:
Yeah, good point, yours is simpler.